libs/gi/formula/src/executors/gen-desc/executor.ts

```diff
@@ -1,14 +1,28 @@
+import type {
+  IConditionalData,
+  IFormulaData,
+} from '@genshin-optimizer/common/formula'
+import type {
+  ArtifactSetKey,
+  CharacterKey,
+  WeaponKey,
+} from '@genshin-optimizer/gi/consts'
 import { workspaceRoot } from '@nx/devkit'
 import { writeFileSync } from 'fs'
 import * as path from 'path'
+import * as prettier from 'prettier'
 import type { GenDescExecutorSchema } from './schema'
-
 // Note:
 // It is important that `data` has NOT been loaded at this point
 // as we are injecting `conditionals` to "collect" the metadata
-import * as prettier from 'prettier'
 import type { entries as Entries } from '../../data'
+import type { Tag } from '../../data/util'
 import { metaList } from '../../data/util'
+
+type SheetKey = CharacterKey | ArtifactSetKey | WeaponKey
```

Comment:
Does this support resonance? Looks like no, as there are conditionals with sheet: `static` and sheet: `reso`